src/main/java/core/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,52 @@
+package core.dao;
+
+import core.db.Storage;
+import core.lib.Dao;
+import core.model.Car;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        Storage.addCar(car);
+        return car;
+    }
+
+    @Override
+    public Car get(Long id) {
+        return getAll().stream()
+                .filter(c -> c.getId().equals(id))
+                .findFirst().get();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return Storage.cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        Car updatedCar = getAll().stream()
+                .filter(c -> c.getId().equals(car.getId()))
+                .findFirst()
+                .get();
```

Comment:
```suggestion
        int indexToReplace = IntStream.range(0. Storage.cars.size())
                .filter(i -> Storage.cars.get().getId().equals(car.getId()))
                .findFirst()
                .getAsInt();
```
and use set below, this way it will be shorter